Applications/FileManager/DirectoryView.cpp

```diff
@@ -121,10 +147,29 @@ void DirectoryView::set_status_message(const String& message)
 
 void DirectoryView::open_parent_directory()
 {
-    model().open(String::format("%s/..", model().path().characters()));
+    auto path = String::format("%s/..", model().path().characters());
+    add_path_to_history(path);
+    model().open(path);
 }
 
 void DirectoryView::refresh()
 {
     model().update();
 }
+
+void DirectoryView::open_previous_directory()
+{
+    if (path_history_position > 0) {
```

Comment:
Small improvement opportunity: could consider disabling the actions when the path is at the start/end. I don't know if we have support for disabled actions, just something to think about :)